cps/kni.c

```diff
@@ -979,6 +982,69 @@ moderror(int err)
 	}
 }
 
+#define SYS_MODULES_ATTR_PATH ("/sys/module/%s/%s")
+
+static int
+get_loaded_kmod_attr(const char *attr, char *val, size_t val_len)
+{
+	FILE *attr_file;
+	char path[128];
+	char line[64];
+	int ret;
+
+	ret = snprintf(path, sizeof(path), SYS_MODULES_ATTR_PATH,
+		KNI_MODULE_NAME, attr);
+	RTE_VERIFY(ret > 0 && ret < (int)sizeof(path));
+
+	attr_file = fopen(path, "r");
+	if (attr_file == NULL) {
+		CPS_LOG(ERR, "Can't open %s: %s\n", path, strerror(errno));
+		return -1;
+	}
+
+	if (fgets(line, sizeof(line), attr_file) != NULL) {
+		size_t len;
+		strncpy(val, line, val_len - 1);
+		val[val_len - 1] = '\0';
+		len = strlen(val);
+		if (len > 0 && val[len - 1] == '\n')
+			val[len - 1] = '\0';
+		ret = 0;
+	} else
+		ret = -1;
+
+	fclose(attr_file);
+	return ret;
+}
+
+static bool
+loaded_kmod_matches_file(void *file, unsigned long len)
+{
+	char kmod_srcver[64], loaded_kmod_srcver[64];
+	int ret;
+
+	ret = get_modinfo_string(file, len, "srcversion",
+		kmod_srcver, sizeof(kmod_srcver));
+	if (ret < 0) {
+		CPS_LOG(ERR, "Unable to fetch srcversion of rte_kni.ko file specified in config\n");
+		return false;
+	}
+
+	ret = get_loaded_kmod_attr("srcversion",
+		loaded_kmod_srcver, sizeof(loaded_kmod_srcver));
+	if (ret < 0) {
+		CPS_LOG(ERR, "Unable to fetch srcversion of rte_kni module already loaded\n");
```

Comment:
Use define `KNI_MODULE_NAME` to avoid having the string `"rte_kni"` in the log message.